geowebcache/azureblob/src/main/java/org/geowebcache/azure/DeleteManager.java

```diff
@@ -281,12 +291,10 @@ public Long call() throws Exception {
                     checkInterrupted();
                     deleteItems(container, response.body().segment(), filter);
                     String marker = response.body().nextMarker();
-                    if (marker != null) {
-                        response =
-                                container.listBlobsFlatSegment(marker, options, null).blockingGet();
-                    } else {
-                        break;
-                    }
+                    // marker will be empty if there is no next page
+                    if (Strings.isNullOrEmpty(marker)) break;
```

Comment:
This is the actual fix for the high number of listBlob. It would just not get out of this loop.